### ISSUE TITLE
.toHaveProperty can not match Template-Literal-Types

### ISSUE THREAD

**Issue by SerkanSipahi** [tstyche/tstyche#360]
For some reason, `.toHaveProperty` can not handle/match `Template Literial Types`. 

### Actual Behavior

```ts
expect<Record<`id_${string}`, boolean>>().type.toHaveProperty('id_foo');
```
```bash
Error: Type 'Record<`id_${string}`, boolean>' does not have property 'id_foo'.
```

### Expected behavior:

```bash
It should not produce any error
```

**Comment by mrazauskas** [comment]
Thanks for good usage example. Indeed index signatures are currently ignored by `.toHaveProperty()`. I did not implement that, because I had no use case around. This has to be fixed, obviously.

**Comment by SerkanSipahi** [comment]
Having these cases is common if you want to make dynamic types accessible for a good developer experience. This feature would benefit especially library developers.

**Comment by mrazauskas** [comment]
This was added and shipped with `tstyche@3.0.0-rc.2`. Please give it a spin.

If nothing critical will happen, I plan to release stable v3 around Wednesday.

**Comment by SerkanSipahi** [comment]
Migrated to v3.0.0-rc.2. It behaves as expected. Thank you.
> This was added and shipped with tstyche@3.0.0-rc.2. Please give it a spin.

**Comment by mrazauskas** [comment]
Great! Thanks for feedback.

**Comment by mrazauskas** [comment]
Shipped with [tstyche@3.0.0](https://github.com/tstyche/tstyche/releases/tag/v3.0.0).
